test/src/test/java/hudson/tasks/BuildTriggerTest.java

```diff
@@ -309,13 +310,6 @@ public void downstreamProjectShouldObserveCompletedParent() throws Exception {
 
         final FreeStyleProject us = j.createFreeStyleProject();
         us.getPublishersList().add(new BuildTrigger("downstream", true));
-        us.getPublishersList().add(new TestNotifier() {
-            @Override
-            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                Thread.sleep(5000); // Build does not complete just after the BuildTrigger
-                return true;
-            }
-        });
```

Comment:
Once confirmed that this change will not break the issue reproducer (021c1b381deec7cab185db4ccaa59ad2b68c4b7a), I am +1 on this.